### PR TITLE
Adx 723 validation workflow

### PR DESCRIPTION
**Pull request by jonathansberry**
This PR significantly improves the validation UI ready for the 22 Estimates.  

- Nicer badges that give clearer information
- Animated throbber in the badge for those jobs still running. 
- Asynchronous update of badges if the validation job is incomplete at the point of loading the page. based on 5s polling of the api. 
- Clear link to the view the validation report "View Error Report" if validation errors or fails. 
- Auto open all error details in the report so that they are not hidden. 
- Clearer layout of validation badges in pages. 

Mali Inputs UNAI

The approach I've taken here is to amend the existing implementation rather than redesign (it would definitely benefit from a redesign!) Validation doesn't have any JS tests in place.  I have updated the python tests that are affected by this commit, though they are currently skipped due to a CKAN2.9 oddity.

### CODE REVIEW

**Comment by tomeksabala** [comment]
Could you have a look at the github actions py tests and either fix them or mark failing test `@pytest.mark.skip(reason="no way of currently testing this")`?

**Comment by jonathansberry** [comment]
@tomeksabala Tests are sorted and passing for this PR.

File: ckanext/validation/assets/js/validation-badge.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        initialize: function () {
+          $.proxyAll(this, /_on/);
+          this._poll();
+          this.options.url = $(this.el).attr('href');
+          $(this.el).removeAttr('href');
+        },
+        _poll: function() {
+            var module = this;
+            $.ajax({
+                url: "/api/3/action/resource_validation_show?resource_id=" + this.options.resource,
```
Comment (tomeksabala) [inline ckanext/validation/assets/js/validation-badge.js:17]: Is it possible to use some sort of `url_for` here not to hard code the API path?
In HTML Jinja we use e.g. `{% url_for 'dataset.search', tags=tag.name %}`

File: ckanext/validation/helpers.py [hunk 2]
```diff
@@ -32,22 +32,38 @@ def get_validation_badge(resource, in_listing=False):
     else:
         status = 'unknown'
 
-    validation_url = url_for(
+    tags = 'href="' + url_for(
```
Comment (tomeksabala) [inline ckanext/validation/helpers.py:35]: this change looks weird.
1. Variable name is kind of misleading (`validation_url` sounded better IMO)
2. Why put `href=` as part of the variable and remove it from line `54`?

**Review by tomeksabala — CHANGES_REQUESTED** [review]

File: ckanext/validation/assets/js/validation-badge.js (hunk 1 above)
Comment (jonathansberry) [reply ckanext/validation/assets/js/validation-badge.js:17]: As stated earlier, documentation is scarce and studying the source code suggests there is not clear way to do this.

**Pull request merged**